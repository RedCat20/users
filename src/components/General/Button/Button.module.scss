.root {
  max-height: 50px;
  display: flex;
  align-items: center;
  justify-content: center;

  padding: 15px;
  border: 1px solid #ccc;
  background: white;
  font-size: 15px;
  border-radius: 15px;
}

.root:hover {
  cursor: pointer;
}

.normal {
  box-shadow: none;
}

.hovered {
  box-shadow: rgba(3, 102, 214, 0.3) 0px 0px 0px 1px;
}