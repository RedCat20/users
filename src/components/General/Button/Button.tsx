import React, {FC, ReactNode, MouseEvent, useState} from 'react';
import styles from './Button.module.scss';

interface Props {
    text?: string;
    children?: ReactNode;
    width?: number;
    isFullWidth?: boolean;
    bgColor?: string;
    textColor?: string;
    marginTop?: number;
    onClickCallback: (e: MouseEvent<HTMLButtonElement>) => void;
}

const STATUS = {
    HOVERED: 'hovered',
    NORMAL: 'normal',
};

const Button: FC<Props> = ({ text,
                             children,
                             width,
                             isFullWidth,
                             bgColor,
                             textColor,
                             onClickCallback,
                             marginTop }) => {

    const [status, setStatus] = useState(STATUS.NORMAL);

    const onMouseEnter = () => {
        setStatus(STATUS.HOVERED);
    };

    const onMouseLeave = () => {
        setStatus(STATUS.NORMAL);
    };

    return (
        <button
            className={`${styles.root} ${styles[status]}`}
            style={{
                width: width ? width : isFullWidth ? '100%' : 'auto',
                background: bgColor ? bgColor : '#fff',
                color: textColor ? textColor : '#000',
                marginTop: marginTop || `${marginTop}px`
            }}
            onClick={onClickCallback}
            onMouseEnter={onMouseEnter}
            onMouseLeave={onMouseLeave}
        >
           <span> {text || children} </span>
        </button>
    );
}

export default Button;
