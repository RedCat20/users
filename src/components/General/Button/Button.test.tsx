import renderer from 'react-test-renderer';
import Button from './Button';

it('changes the class when hovered', () => {
    const component = renderer.create(
        <Button onClickCallback={() => {}}>Facebook</Button>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
        if (!tree) return;
        if ("props" in tree) {
            tree.props.onMouseEnter();
        }
    });
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    renderer.act(() => {
        if (!tree) return;
        if ("props" in tree) {
            tree.props.onMouseLeave();
        }
    });
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});