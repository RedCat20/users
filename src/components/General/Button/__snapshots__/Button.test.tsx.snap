// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<button
  className="root normal"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "#fff",
      "color": "#000",
      "marginTop": "undefinedpx",
      "width": "auto",
    }
  }
>
  <span>
     
    Facebook
     
  </span>
</button>
`;

exports[`changes the class when hovered 2`] = `
<button
  className="root hovered"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "#fff",
      "color": "#000",
      "marginTop": "undefinedpx",
      "width": "auto",
    }
  }
>
  <span>
     
    Facebook
     
  </span>
</button>
`;

exports[`changes the class when hovered 3`] = `
<button
  className="root normal"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "background": "#fff",
      "color": "#000",
      "marginTop": "undefinedpx",
      "width": "auto",
    }
  }
>
  <span>
     
    Facebook
     
  </span>
</button>
`;
